lib/srv/debug/handlers.go

```diff
@@ -0,0 +1,109 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package debug
+
+import (
+	"fmt"
+	"io"
+	"log/slog"
+	"net/http"
+	"net/http/pprof"
+
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+)
+
+// LogLeveler defines a struct that can retrieve and set log levels.
+type LogLeveler interface {
+	// GetLogLevel returns the current log level.
+	GetLogLevel() slog.Level
+	// SetLogLevel sets the log level.
+	SetLogLevel(slog.Level)
+}
+
+// NewServeMux returns a http mux that handles all the debug service endpoints.
+func NewServeMux(logger *slog.Logger, leveler LogLeveler) *http.ServeMux {
+	mux := http.NewServeMux()
+	mux.HandleFunc(PProfEndpointsPrefix+"cmdline", pprofMiddleware(logger, "cmdline", pprof.Cmdline))
+	mux.HandleFunc(PProfEndpointsPrefix+"profile", pprofMiddleware(logger, "profile", pprof.Profile))
+	mux.HandleFunc(PProfEndpointsPrefix+"symbol", pprofMiddleware(logger, "symbol", pprof.Symbol))
+	mux.HandleFunc(PProfEndpointsPrefix+"trace", pprofMiddleware(logger, "trace", pprof.Trace))
+	mux.HandleFunc(PProfEndpointsPrefix+"{profile}", func(w http.ResponseWriter, r *http.Request) {
+		pprofMiddleware(logger, r.PathValue("profile"), pprof.Index)(w, r)
+	})
```

Comment:
I initially did this, but it seems we're misusing the serve mux in this case, defining the routes inside a handler instead of on the mux itself.

Also, having all definitions in one place seems more straightforward to read (even with the middleware itself). We can look at one location and tell exactly what routes are being served by this service (which I find helpful when reading the source code).